plugins/module_utils/network/sonic/config/system/system.py

```diff
@@ -225,27 +225,37 @@ def _state_replaced(self, want, have):
         :returns: the commands necessary to migrate the current configuration
                   to the desired configuration
         """
+        commands = []
+        requests = []
+        merged_commands = []
+        merged_requests = []
+
         new_want = self.patch_want_with_default(want, ac_address_only=True)
         replaced_config = self.get_replaced_config(have, new_want)
         if replaced_config:
             requests = self.get_delete_all_system_request(replaced_config)
-            send_requests(self._module, requests)
-            commands = new_want
-        else:
-            diff = get_diff(new_want, have)
-            commands = diff
-            if not commands:
-                commands = []
-
-        requests = []
-
-        if commands:
-            requests = self.get_create_system_request(have, commands)
-
             if len(requests) > 0:
-                commands = update_states(commands, "replaced")
+                commands = update_states(replaced_config, "deleted")
+                if 'hostname' in commands or 'interface_naming' in commands:
+                    # All existing config is to be deleted.
+                    merged_commands = new_want
```

Comment:
If 'hostname' in commands or 'interface_naming' in commands, but they are same as those in have, they should not be merged.

Reply:
This is inside a block where it has been determined that something got deleted. Anything in "commands" at this point is an attribute that was deleted. Since it was deleted, it needs to be re-configured.